big-red-shuttle/StopsViewController.swift

```diff
@@ -101,5 +123,186 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
+        directionsButton.setTitle("Directions", for: .normal)
+        directionsButton.setTitleColor(.lightGray, for: .normal)
+        directionsButton.addTarget(self, action: #selector(directionsButtonPressed), for: .touchUpInside)
+        directionsButton.center.y = midHeight
+        
+        let directionImage = UIImageView()
+        directionImage.image = UIImage(cgImage: #imageLiteral(resourceName: "arrow").cgImage!, scale: 1.0, orientation: UIImageOrientation.left)
+        directionImage.frame = CGRect(x: popUpWidth - 25, y: 14.25, width: 9, height: 16)
+        directionImage.center.y = midHeight
+        
+        let locationLabelFrame = CGRect(x: 40, y: 11.75, width: popUpWidth * 0.50, height: 26)
+        let locationLabel = UILabel(frame: locationLabelFrame)
+        locationLabel.text = selectedStop.name
+        locationLabel.font = .systemFont(ofSize: 16, weight: UIFontWeightSemibold)
+        locationLabel.numberOfLines = 2
+        locationLabel.center.y = midHeight
+        locationLabel.sizeToFit()
+        
+        let locationImage = UIImageView(image: #imageLiteral(resourceName: "location"))
+        locationImage.frame = CGRect(x: 10, y: 18.5, width: 20, height: 26)
+        locationLabel.center.y = midHeight
+        
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.itemSize = CGSize(width: 70, height: 30)
+        
+        let collectionView = UICollectionView(frame: CGRect(x: 8, y: 64, width: popUpViewFrame.width - 8, height: 40), collectionViewLayout: layout)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.register(CustomTimeCell.self, forCellWithReuseIdentifier: "Cell")
+        collectionView.backgroundColor = .clear
+        collectionView.showsHorizontalScrollIndicator = false
+        
+        let middleBorder = CALayer()
+        middleBorder.frame = CGRect(x: 0, y: 64, width: popUpViewFrame.width, height: 1)
+        middleBorder.backgroundColor = UIColor(red: 227/255, green: 229/255, blue: 233/255, alpha: 1.0).cgColor
+        
+        popUpView.addSubview(collectionView)
+        popUpView.addSubview(directionsButton)
+        popUpView.addSubview(directionImage)
+        popUpView.addSubview(locationLabel)
+        popUpView.addSubview(locationImage)
+        popUpView.layer.addSublayer(middleBorder)
+        view.addSubview(popUpView)
+        
+        UIView.animate(withDuration: 0.2, delay: 0.0, usingSpringWithDamping: 0.9, initialSpringVelocity: 0.5, options: .curveEaseInOut, animations: {
+            self.popUpView.frame = CGRect(x: 12, y: viewHeight - 120 , width: popUpWidth, height: 106)
+        }, completion: { _ in
+            
+            if (UserDefaults.standard.value(forKey: "nudgeCount") as! Int) < 3 && UserDefaults.standard.value(forKey: "didFireNudge") as! Bool {
```

Comment:
Store `(UserDefaults.standard.value(forKey: "nudgeCount") as! Int)` in a variable so you can use it in line 253 as well without calling `UserDefaults` again